sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -1093,6 +1126,16 @@ class HistGradientBoostingRegressor(RegressorMixin, BaseHistGradientBoosting):
 
         .. versionadded:: 0.23
 
+    interaction_cst : list of lists or sets of int, default=None
+        Each inner list or set specifies the feature indices that are allowed
+        to interact with each other, meaning splits in child nodes are only
+        allowed in those sets. For instance, `[{0, 1}, {2, 3, 4}]` says that
+        a single tree is either only split on features 0 and 1 or only on
```

Comment:
If I understand the code correctly, the constrains don't constrain the whole tree, they constrain single branches? I would just change into 

> For instance, `[{0, 1}, {2, 3, 4}]` says that a given branch of a tree is either only split...

We can indicate that a branch is a path from the root the a leave, but I'm not sure it's needed